application/F3DConfigFileTools.cxx

```diff
@@ -177,12 +177,34 @@ F3DOptionsTools::OptionsEntries F3DConfigFileTools::ReadConfigFiles(const std::s
       continue;
     }
 
-    // For each config "pattern"
-    for (const auto& configBlock : json.items())
+    // For each config block in the main array
+    for (const auto& configBlock : json)
     {
-      // Add each config entry into an option dict
+      // Recover match if any
+      std::string match;
+      try
+      {
+        match = configBlock.at("match");
+      }
+      catch (nlohmann::json::exception&)
```

Comment:
```suggestion
      catch (nlohmann::json::out_of_range&)
```
more specific exception for expected missing key so it doesn't ignore actual issues unintentionally